src/Sarif/ConsoleLogger.cs

```diff
@@ -10,17 +10,46 @@ namespace Microsoft.CodeAnalysis.Sarif
 {
     public class ConsoleLogger : IAnalysisLogger
     {
-        public ConsoleLogger(bool verbose)
+        public ConsoleLogger(bool verbose, string toolName)
         {
             Verbose = verbose;
+            _toolName = toolName.ToUpperInvariant();
         }
 
+        private StringBuilder _capturedOutput;
+
+        public bool CaptureOutput { get; set; }
+
+        public string CapturedOutput
+        {
+            get
+            {
+                if (_capturedOutput != null)
+                {
+                    return _capturedOutput.ToString();
+                }
+                return null;
+            }
+        }
+
+        private string _toolName;
+
         public bool Verbose { get; set; }
 
+        private void WriteLineToConsole(string text = null)
+        {
+            Console.WriteLine(text);
+
+            if (CaptureOutput)
+            {
+                _capturedOutput = _capturedOutput ?? new StringBuilder();
```

Comment:
>_capturedOutput [](start = 16, length = 15)

Can't wait for C#8 where this is just `_capturedOutput ??= new StringBuilder();`

Reply:
nice!

---
In reply to: [303172128](https://github.com/microsoft/sarif-sdk/pull/1574#discussion_r303172128) [](ancestors = 303172128)